### DISCUSSION TITLE
Change table state variable for grouping

### DISCUSSION THREAD

**Discussion by VinitraMk** [TanStack/table#2405] [General]
I am trying to implement a scenario wherein I need to change the columns to group by. Initially, I am passing an empty array to initialState.groupBy prop. But on click of a button I need to update the groupBy state to an array of strings. I am using react table v7. Is there anything in this version that can help me achieve this functionality? Is it possible to control the groupBy state of the table in v7?

**Comment by 07harish** [comment]
Yes it is possible

- You can dispatch (instance.dispatch) the action & control your state in **useControlledState** or **StateReducer** to handle state

- toggleGroupBy is available in the instance & column level

You get the instance when you pass a function to useTable options.
